kafka.go

```diff
@@ -119,24 +151,99 @@ func newConfig() *sarama.Config {
 	return config
 }
 
-func (a *KafkaAdapter) formatMessage(message *router.Message) (*sarama.ProducerMessage, error) {
+/* *
+	Strava Logstash Message
+	Annotated with the marathon/mesos information that's only available on the particular instance
+	https://github.com/gliderlabs/logspout/blob/5abc836e8cabcaebd862d981fa7fbea8798ff4d0/router/types.go#L52
+	Logspout Message Struct
+
+	// Message is a log messages
+	type Message struct {
+		Container *docker.Container // the fsouza docker container
+		Source    string // stdout, stdin etc
+		Data      string // the actual data
+		Time      time.Time
+	}
+* */
+func (a *KafkaAdapter) formatToLogstashMessage(message *router.Message) (*sarama.ProducerMessage, error) {
 	var encoder sarama.Encoder
-	if a.tmpl != nil {
-		var w bytes.Buffer
-		if err := a.tmpl.Execute(&w, message); err != nil {
-			return nil, err
-		}
-		encoder = sarama.ByteEncoder(w.Bytes())
-	} else {
-		encoder = sarama.StringEncoder(message.Data)
+
+	// ignore the template variable
+	js, err := createLogstashMessage(message)
+
+	if err != nil {
+		return nil, err
 	}
 
+	encoder = sarama.ByteEncoder(js)
+
+	// Note: ProducerMessage also has a "Timestamp" field
+	// https://github.com/Shopify/sarama/blob/65f0fec86aabe011db77ad641d31fddf14f3ca41/async_producer.go
 	return &sarama.ProducerMessage{
 		Topic: a.topic,
 		Value: encoder,
 	}, nil
 }
 
+func createLogstashMessage(message *router.Message) ([]byte, error) {
+	imageName, imageTag := splitImage(message.Container.Config.Image)
+	host := envValue("HOST", message.Container.Config.Env)
+
+	if host == nil {
+		host = &message.Container.Config.Hostname
+	}
+
+	var data map[string]interface{}
+	json.Unmarshal([]byte(message.Data), &data) // not sure if this is necessary?
```

Comment:
I think this is only necessary if you want to get "valid logback looking json", whatever that means?
https://github.com/strava/logspout-redis-logstash/blob/55ddd4dd5b8790b8e8ea494982af6225eb0e958b/redis.go#L292-L294
 if you aren't going to have that check then I think you can remove it

Reply:
if you do keep this, you're swallowing potential errors
```suggestion
	if err := json.Unmarshal([]byte(message.Data), &data); err != nil {
	  return nil
	}
```

Reply:
thanks, conner!